### ISSUE TITLE
Auto complete of props in MDC

### ISSUE THREAD

**Issue by Atinux** [nuxtlabs/vscode-mdc#24]
It would be great to have the possibility to support the autocompletion of props in the editor.

**Comment by Atinux** [comment]
I guess we could leverage something similar to what @tahul has done in Pinceau: https://github.com/Tahul/pinceau/blob/main/volar/index.js

By using nuxt-component-meta under the hood?

**Comment by mrwwalmsley** [comment]
I was wanting to implement autocomplete for my React project at work.
I was planning to have auto complete of props as well as directive names by reading the filenames of MDC directive components.

Would it be possible to have a configuration file that can specify all the props available for each directive so that this could work in a React as well as Vue project?

**Comment by Atinux** [comment]
Did you make the MDC works with React @mrwwalmsley ?

**Comment by mrwwalmsley** [comment]
Yes, we have been using MDC in a React project for a couple of years -- built on Remark -- with a few custom extensions.

We had own custom editor & syntax highlighting that was not as good as the Nuxt3 syntax highlighter. When Nuxt3 added MDC I started upgrading my Nuxt side projects from Nuxt2 to Nuxt3 using MDC.  After that we started testing the Nuxt plugin for writing our React content in VsCode.

Our MDC implementation has a few extras such as include statements:

```
:::define[template_name]
Hello {{name}}!
:::

:::include[template_name]
name='Mike'
:::

:::include[template_name]
name='Sebastian'
:::
```

If the includes are not defined locally, then they are loaded from our content DB.

**Comment by mrwwalmsley** [comment]
I've been using this for a Nuxt website I have been rebuilding from Options+Nuxt2 to Composition+Nuxt3.  It would be amazing to get autocomplete and syntax highlighting for attributes as well as slot names:

![image](https://github.com/user-attachments/assets/90ba5ab6-175a-4960-a740-45dd925d1ddd)

And ideally would give a red squiggly underline for slots or attributes that don't exist.
This would make it much easier for content writers.

How hard to do this? Anything I can do to help?

**Comment by adamdehaven** [comment]
@atinux I have suggestions working for MDC block components in my project (I fetch the component data from a Nitro endpoint in my app). I'm planning to add support for suggesting & validating props, but haven't gotten there yet.

I'm not sure where to source the component data from for this repo, but I can contribute the rest of the logic if someone wants to assist

**Comment by atinux** [comment]
I think @farnabaz would be the best to help on this, I guess we shall somehow leverage nuxt-component-meta for it 🤔

**Comment by adamdehaven** [comment]
See https://github.com/nuxtlabs/vscode-mdc/pull/43
